src/RootCloseWrapper.js

```diff
@@ -54,7 +60,22 @@ export default class RootCloseWrapper extends React.Component {
   }
 
   render() {
-    return React.Children.only(this.props.children);
+    // Wrap the child in a new element, so the child won't have to handle
+    // potentially combining multiple onClick listeners.
+    const child = React.cloneElement(
+      React.Children.only(this.props.children),
+      {ref: component => this._wrappedElement = component}
+    );
+
+    return (
+      <div onClick={suppressRootClose}>
+        {child}
```

Comment:
I didn't want to just pass through `onClick` because I was concerned about the case of custom overlay components, or ones that already have `onClick` specified.

I didn't want to add a native handler because I was concerned about the overlay DOM node getting replaced by various mechanisms.

The extra element does not mess with the layout (there already are wrapping elements for ovelrays in new layers anyway), though I needed the `getWrappedElement` workaround below to get size calculations to work correctly.

I agree about your point with refs - I misread this part:
>Note: React.cloneElement(child, { ref: 'newRef' }) DOES override the ref so it is still not possible for two parents to have a ref to the same child, unless you use callback-refs.
to mean that `cloneElement` would do something smart with callback-refs, but I can see that it doesn't, so I need to re-write this bit.